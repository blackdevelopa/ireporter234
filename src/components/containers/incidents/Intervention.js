import React, {Component } from 'react';
import classes from './Incident.css';
import { connect } from 'react-redux';
import Incident from '../../incident/incident';
import Navbar from '../../navbar/Navbar';
import SwitchNav from '../../navbar/switchNav/SwitchNav';
import { withRouter } from 'react-router-dom';
import { fetchInterventionIncident } from '../../../store/actions/incident/intervention-dispatcher';

class Intervention extends Component {

  componentDidMount() {
    console.log('hi')
    this.props.fetchInterventionIncident()
  }
  render () {
  return (
    <div className={classes.main}>
      <Navbar name="Profile" />
      <SwitchNav />
      <Incident />
      <Incident />
      <Incident />
    </div>
  )
}
}

const mapStateToProps = state => ({
  sucess: state.intervention.success
})

export default connect(mapStateToProps, {fetchInterventionIncident})(withRouter(Intervention));