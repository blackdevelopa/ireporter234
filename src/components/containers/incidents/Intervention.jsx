/* eslint-disable react/prop-types */
/* eslint-disable react/destructuring-assignment */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { Container, Card } from 'semantic-ui-react';
import classes from './Incident.css';
import Navbar from '../../navbar/Navbar';
import SwitchNav from '../../navbar/switchNav/SwitchNav';
import { fetchAllIncident } from '../../../redux/actions/incident/incident';

class Intervention extends Component {
  componentDidMount() {
    this.props.fetchAllIncident('interventions');
  }

  render() {
    const interventionInfo = this.props.intervention.map(data => {
      const descriptionText = (
        <div className={classes.Ellipsis}>{data.comment}</div>
      );
      return {
        key: data.id,
        image: data.images,
        header: data.location,
        extra: data.status,
        description: descriptionText,
        meta: data.createdon.substr(0, 10),
      };
    });

    return (
      <div className={classes.main}>
        <Navbar name="Profile" />
        <SwitchNav />
        <div className={classes.Group}>
          <Container>
            <Card.Group
              href="#card-example-link-card"
              items={interventionInfo}
              centered
            />
          </Container>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  intervention: state.incident.incident,
});

export default connect(
  mapStateToProps,
  { fetchAllIncident }
)(withRouter(Intervention));
