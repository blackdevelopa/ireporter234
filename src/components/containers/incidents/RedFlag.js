import React, {Component } from 'react';
import classes from './Incident.css';
import { connect } from 'react-redux';
import Navbar from '../../navbar/Navbar';
import SwitchNav from '../../navbar/switchNav/SwitchNav';
import { fetchRedflagIncident } from '../../../store/actions/incident/redflag-dispatchers';
import { withRouter } from 'react-router-dom';
import pic from '../../../assets/download.png';
import { Card } from 'semantic-ui-react';

class Redflag extends Component {

  componentDidMount() {
    this.props.fetchRedflagIncident()
  }
  render () {
  const redflagInfo = this.props.redflag.map(data => {
    return {
        image: {pic},
        header: data.location,
        description: data.comment,
        meta: data.createdon,
      }
    
  })

  return (
    <div className={classes.main}>
      <Navbar name="Profile" />
      <SwitchNav />
      <Card.Group items={redflagInfo} />
    </div>
  )
}
}


const mapStateToProps = state => ({
  redflag: state.redflag.redflag
})

export default connect(mapStateToProps, {fetchRedflagIncident})(withRouter(Redflag));