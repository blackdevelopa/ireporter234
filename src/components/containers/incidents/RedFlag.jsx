/* eslint-disable react/forbid-prop-types */
/* eslint-disable no-shadow */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import { Container, Card } from 'semantic-ui-react';
import classes from './Incident.css';
import Navbar from '../../navbar/Navbar';
import SwitchNav from '../../navbar/switchNav/SwitchNav';
import { fetchAllIncident } from '../../../redux/actions/incident/incident';
import ViewCard from './viewSingleIncident';

export class Redflag extends Component {
  componentDidMount() {
    const { fetchAllIncident } = this.props;
    fetchAllIncident('red-flags');
  }

  firstbtnclick = () => {
    const { history } = this.props;
    history.push('/new-red-flag');
  };

  secondbtnclick = () => {
    const { history } = this.props;
    history.push('/interventions');
  };

  render() {
    const { redflag } = this.props;
    return (
      <div className={classes.main}>
        <Navbar
          firstbtnclick={this.firstbtnclick}
          secondbtnclick={this.secondbtnclick}
          firstbtn="Add New Redflag"
          secondbtn="Intervention"
          access="true"
        />
        <SwitchNav />
        <div className={classes.Group}>
          <Container>
            <Card.Group centered>
              <ViewCard data={redflag} type="red-flags" />
            </Card.Group>
          </Container>
        </div>
      </div>
    );
  }
}

Redflag.propTypes = {
  fetchAllIncident: PropTypes.func.isRequired,
  redflag: PropTypes.array.isRequired,
  history: PropTypes.shape({}).isRequired,
};

const mapStateToProps = state => ({
  redflag: state.incident.incident,
});

export default connect(
  mapStateToProps,
  { fetchAllIncident }
)(withRouter(Redflag));
