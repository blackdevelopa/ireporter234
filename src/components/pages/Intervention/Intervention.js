import React, { Component } from 'react'

class Intervention extends Component {
  render() {
    return (
      <div>
        <h2>Intervention</h2>
      </div>
    )
  }
}

export default Intervention
