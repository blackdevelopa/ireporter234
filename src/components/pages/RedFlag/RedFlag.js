import React, {Component} from 'react';
// import Navbar from '../../navbar/Navbar';

class RedFlag extends Component {
  render() {
    return (
      <div>
      {/* <div className="side_nav">
        <div className="title">
          <img src="../img/gloves.jpeg" alt="cardimage" />
        </div>
      </div>
      <div className="side_body">
        <p className="user_text"><Strong>Name: </Strong>Sylva Elendu</p>
        <p className="user_text"><strong>Profession: </strong>Student</p>
        <p className="user_text"><strong>email: </strong>john@dummy.com</p>
        <p className="user_text"><strong>About Me: </strong>I am StongFoodie</p>
      </div> */}
      <h2>Redflag</h2>
      </div>
    )
  }
}

export default RedFlag;
