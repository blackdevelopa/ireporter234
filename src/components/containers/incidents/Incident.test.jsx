import React from 'react';
import { shallow } from 'enzyme';
import { Redflag } from './RedFlag';
import { Intervention } from './Intervention';

const props = {
  fetchAllIncident: jest.fn(),
  redflag: {},
  intervention: {},
};

describe('<IncidentPage />', () => {
  const redflagWrapper = shallow(<Redflag {...props} />);
  const interventionWrapper = shallow(<Intervention {...props} />);
  it('should render without crashing', () => {
    expect(redflagWrapper).toMatchSnapshot();
    expect(interventionWrapper).toMatchSnapshot();
  });
});
