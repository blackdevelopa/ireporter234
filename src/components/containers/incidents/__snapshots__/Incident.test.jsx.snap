// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IncidentPage /> should render without crashing 1`] = `
<div>
  <Navbar
    access="true"
    firstbtn="Add New Redflag"
    firstbtnclick={[Function]}
    secondbtn="Intervention"
    secondbtnclick={[Function]}
  />
  <switchNav />
  <div>
    <Container>
      <CardGroup
        centered={true}
      >
        <ViewCard
          data={Object {}}
          type="red-flags"
        />
      </CardGroup>
    </Container>
  </div>
</div>
`;

exports[`<IncidentPage /> should render without crashing 2`] = `
<div>
  <Navbar
    access="true"
    firstbtn="Add New Intervention"
    firstbtnclick={[Function]}
    secondbtn="Redflag"
    secondbtnclick={[Function]}
  />
  <switchNav />
  <div>
    <Container>
      <CardGroup
        centered={true}
      >
        <ViewCard
          data={Object {}}
          type="interventions"
        />
      </CardGroup>
    </Container>
  </div>
</div>
`;
